lib/commands/command-install.bash

```diff
@@ -1,243 +1,9 @@
-    local version="${version_info[1]}"
-  else
-    local install_type="version"
-
-    if [ "${version_info[0]}" = "latest" ]; then
-      local version
-      version=$(asdf latest "$plugin_name" "${version_info[1]}")
-      full_version=$version
-    else
-      local version="${version_info[0]}"
-    fi
-  fi
-
-  local install_path
-  install_path=$(get_install_path "$plugin_name" "$install_type" "$version")
-  local download_path
-  download_path=$(get_download_path "$plugin_name" "$install_type" "$version")
-  local concurrency
-  concurrency=$(get_concurrency)
-  trap 'handle_cancel $install_path' INT
-
-  if [ -d "$install_path" ]; then
-    printf "%s %s is already installed\\n" "$plugin_name" "$full_version"
-  else
-
-    if [ -f "${plugin_path}/bin/download" ]; then
-      # Not a legacy plugin
-      # Run the download script
-      (
-        # shellcheck disable=SC2030
-        export ASDF_INSTALL_TYPE=$install_type
-        # shellcheck disable=SC2030
-        export ASDF_INSTALL_VERSION=$version
-        # shellcheck disable=SC2030
-        export ASDF_INSTALL_PATH=$install_path
-        # shellcheck disable=SC2030
-        export ASDF_DOWNLOAD_PATH=$download_path
-        mkdir "$download_path"
-        asdf_run_hook "pre_asdf_download_${plugin_name}" "$full_version"
-        "${plugin_path}"/bin/download
-      )
-    fi
-
-    local download_exit_code=$?
-    if [ $download_exit_code -eq 0 ]; then
-      (
-        # shellcheck disable=SC2031
-        export ASDF_INSTALL_TYPE=$install_type
-        # shellcheck disable=SC2031
-        export ASDF_INSTALL_VERSION=$version
-        # shellcheck disable=SC2031
-        export ASDF_INSTALL_PATH=$install_path
-        # shellcheck disable=SC2031
-        export ASDF_DOWNLOAD_PATH=$download_path
-        # shellcheck disable=SC2031
-        export ASDF_CONCURRENCY=$concurrency
-        mkdir "$install_path"
-        asdf_run_hook "pre_asdf_install_${plugin_name}" "$full_version"
-        "${plugin_path}"/bin/install
-      )
-    fi
-
-    local install_exit_code=$?
-    if [ $install_exit_code -eq 0 ] && [ $download_exit_code -eq 0 ]; then
-      # Remove download directory if --keep-download flag or always_keep_download config setting are not set
-      always_keep_download=$(get_asdf_config_value "always_keep_download")
-      if [ ! "$keep_download" = "true" ] && [ ! "$always_keep_download" = "yes" ] && [ -d "$download_path" ]; then
-        rm -r "$download_path"
-      fi
-
-      asdf reshim "$plugin_name" "$full_version"
-
-      asdf_run_hook "post_asdf_install_${plugin_name}" "$full_version"
-    else
-      handle_failure "$install_path"
-    fi
-  fi
-}
+# shellcheck source=lib/functions/versions.bash
+. "$(dirname "$(dirname "$0")")/lib/functions/versions.bash"
+# shellcheck source=lib/commands/reshim.bash
+. "$(dirname "$ASDF_CMD_FILE")/reshim.bash"
+# shellcheck source=lib/functions/installs.bash
+. "$(dirname "$(dirname "$0")")/lib/functions/installs.bash"
```

Comment:
Code was re-arranged and calls to `asdf <cmd>` were replaced with the corresponding function calls, but no other code was changed. The diff is pretty big but most of it is moving code into the new function files.